src/o365/spo/commands/listitem/listitem-record-undeclare.ts

```diff
@@ -0,0 +1,305 @@
+    telemetryProps.listId = typeof args.options.listId !== 'undefined';
+    telemetryProps.listTitle = typeof args.options.listTitle !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    const listIdArgument = args.options.listId || '';
+    const listTitleArgument = args.options.listTitle || '';
+    let siteAccessToken: string = '';
+    const listRestUrl: string = (args.options.listId ?
+      `${args.options.webUrl}/_api/web/lists(guid'${encodeURIComponent(listIdArgument)}')`
+      : `${args.options.webUrl}/_api/web/lists/getByTitle('${encodeURIComponent(listTitleArgument)}')`);
+
+
+    let formDigestValue: string = '';
+    let environmentListId: string = '';
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise | Promise<any> => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}.`);
+          cmd.log(``);
+          cmd.log(`auth object:`);
+          cmd.log(auth);
+        }
+        if (this.verbose) {
+          cmd.log(`Getting list id...`);
+        }
+
+          const listRequestOptions: any = {
+            url: `${listRestUrl}/id`,
+            headers: Utils.getRequestHeaders({
+              authorization: `Bearer ${siteAccessToken}`,
+              'accept': 'application/json;odata=nometadata'
+            }),
+            json: true
+          };
+
+          if (this.debug) {
+            cmd.log('Executing web request for list id...');
+            cmd.log(listRequestOptions);
+            cmd.log('');
+          }
+          return request.get(listRequestOptions)
+      
+      })
+      .then((dataReturned: any): request.RequestPromise | Promise<void> => {
+        if (dataReturned) {
+          environmentListId = dataReturned.value;
+          if (this.debug) {
+            cmd.log(`data returned[0]:`);
+            cmd.log(dataReturned);
+            cmd.log(`Retrieved list id ${environmentListId}.`);
+          }
+        }
+        if (this.debug) {
+          cmd.log(`getting request digest for request`);
+        }
+        return this.getRequestDigestForSite(args.options.webUrl, siteAccessToken, cmd, this.debug);
+      })
+      .then((res: ContextInfo): Promise<string> => {
+        if (this.debug) {
+          cmd.log('Response:')
+          cmd.log(res);
+          cmd.log('');
+        }
+        if (this.verbose) {
+          cmd.log(`Undeclare list item  as a record in list ${args.options.listId || args.options.listTitle} in site ${args.options.webUrl}...`);
+        }
+        formDigestValue = res['FormDigestValue'];
```

Comment:
Instead of using `res['FormDigestValue']` we should use `res.FormDigestValue` which is less error-prone